integration/cri-o/cri-o.sh

```diff
@@ -56,7 +56,27 @@ fi
 # This will allow us to run tests with at least 2 different
 # storage drivers.
 if [ "$ID" == "ubuntu" ] && [ "$MAJOR" -ge 17 ]; then
-	export CRIO_STORAGE_DRIVER_OPTS="--storage-driver overlay"
+	export STORAGE_OPTIONS="--storage-driver overlay"
+fi
+
+
+# If we are testing a CRI-O PR in a CI environment,
+# use devicemapper driver using a block device.
+# This is needed since support for loopback devices
+# is being deprecated.
+# More info: https://github.com/kubernetes-incubator/cri-o/pull/1574
+#            https://github.com/containers/storage/pull/80/commits/da1e7e5d28da7d0a1e6ac0d4a9e647bee72282e2
+if [ "$ghprbGhRepository" == "${crio_repository/github.com\/}" ] && [ "$CI" == true ];then
+	# block device attached to the Azure VM where we run
+	# the kata-containers CI
+	block_device=/dev/sdb
```

Comment:
To go further into the protection of a developer machine, we should start at `/dev/sde` and increment with `/dev/sdf`, `/dev/sdg`, ... until we find a path/device that does not exist. This way, we make sure we don't override an existing device.

Reply:
@sboeuf `/dev/sdb` is the device that will be available on the Azure machines. I don't think we can manage to go until `/dev/sde` unless we attach more disks to the VM. 
What I was thinking is that I can wipe the partitions of `/dev/sdb` in the VM startup script of the Jenkins setup. This way we do not mess with developer machines.

Reply:
+10000000 for lots more paranoia in this scenario.

Can you also add a check so that if `KATA_DEV_MODE` is set (even if `CI` is set), we do *NOT* do this to be more paranoid about dev mode:

- https://github.com/kata-containers/tests#developer-mode

For extra safety, we could potentially only run this if we're in a VM. I'm not sure how we check this reliably though since the following only seems to work on `x86_64`:

```
$ egrep -m1 "flags.*:.*\<hypervisor\>" /proc/cpuinfo
```

Reply:
@jodh-intel I can also check that `sdb` does not have a partition before running the tests. I'll add the  `KATA_DEV_MODE` check to not run the tests if it is set